src/saveload/de.rs

```diff
@@ -0,0 +1,179 @@
+    pd: PhantomData<(M, E, T)>,
+}
+
+impl<'de, 'a, 'b: 'a, M, E, T> DeserializeSeed<'de> for DeserializeEntity<'a, 'b, M, E, T>
+where
+    M: Marker,
+    E: Error,
+    T: Components<M::Identifier, E>,
+{
+    type Value = ();
+    fn deserialize<D>(self, deserializer: D) -> Result<(), D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let DeserializeEntity {
+            entities,
+            storages,
+            markers,
+            allocator,
+            ..
+        } = self;
+        let data = EntityData::<M, E, T>::deserialize(deserializer)?;
+        let entity = allocator.marked(data.marker.id(), entities, markers);
+        markers
+            .get_mut(entity)
+            .ok_or("Allocator is broken")
+            .map_err(de::Error::custom)?
+            .update(data.marker);
+        let ids = |marker: M::Identifier| Some(allocator.marked(marker, entities, markers));
+        T::load(entity, data.components, storages, ids).map_err(de::Error::custom)?;
+        Ok(())
+    }
+}
+
+
+/// Wrapper for `Entities` and tuple of `WriteStorage`s that implements `serde::Deserialize`
+struct VisitEntities<'a, 'b: 'a, M: Marker, E: Error, T: Components<M::Identifier, E>> {
+    entities: &'a Entities<'b>,
+    storages: &'a mut <T as Storages<'b>>::WriteStorages,
+    markers: &'a mut WriteStorage<'b, M>,
+    allocator: &'a mut FetchMut<'b, M::Allocator>,
+    pd: PhantomData<(M, E, T)>,
+}
+
+impl<'de, 'a, 'b: 'a, M, E, T> Visitor<'de> for VisitEntities<'a, 'b, M, E, T>
+where
+    M: Marker,
+    E: Error,
+    T: Components<M::Identifier, E>,
+{
+    type Value = ();
+
+    fn expecting(&self, formatter: &mut Formatter) -> fmt::Result {
+        write!(formatter, "Sequence of serialized entities")
+    }
+
+    fn visit_seq<A>(mut self, mut seq: A) -> Result<(), A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        while let Some(()) = seq.next_element_seed(DeserializeEntity {
+            entities: self.entities,
+            storages: self.storages,
+            markers: self.markers,
+            allocator: self.allocator,
+            pd: self.pd,
+        })? {}
+
+        Ok(())
+    }
+}
+
+
+/// Deserialize entities
+pub fn deserialize<'a, 'de, D, M, E, T>(
+    entities: &Entities<'a>,
+    storages: &mut <T as Storages<'a>>::WriteStorages,
+    markers: &mut WriteStorage<'a, M>,
+    allocator: &mut FetchMut<'a, M::Allocator>,
+    deserializer: D,
+) -> Result<(), D::Error>
+where
+    M: Marker,
+    E: Error,
+    T: Components<M::Identifier, E>,
+    D: Deserializer<'de>,
+{
+    deserializer.deserialize_seq(VisitEntities::<M, E, T> {
+        entities: entities,
+        storages: storages,
+        markers: markers,
+        allocator: allocator,
+        pd: PhantomData,
+    })
+}
+
+
+/// `DeerializeSeed` implementation for `World`
+//#[derive(SystemData)]
+pub struct WorldDeserialize<'a, M: Marker, E: Error, T: Components<M::Identifier, E>> {
```

Comment:
You shouldn't create unnecessary bounds on the struct itself, e.g. `E: Error` should be removed here.